connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -237,32 +237,44 @@ public ConnectorStateInfo connectorStatus(String connName) {
                 status.workerId(), status.trace());
     }
 
+    protected Map<String, ConfigValue> validateBasicConnectorConfig(Connector connector,
+                                                                    ConfigDef configDef,
+                                                                    Map<String, String> config) {
+        return configDef.validateAll(config);
+    }
 
     @Override
-    public ConfigInfos validateConfigs(String connType, Map<String, String> connectorConfig) {
+    public ConfigInfos validateConnectorConfig(Map<String, String> connectorConfig) {
+        String connType = connectorConfig.get(ConnectorConfig.CONNECTOR_CLASS_CONFIG);
+        if (connType == null)
+            throw new IllegalArgumentException("Connector config " + connectorConfig + " contains no connector type");
```

Comment:
Is `IllegalArgumentException` what we want given this is going to get thrown to a Jersey handler? Presumably we'd want an exception type that will get converted to a useful error message by an exception mapper?